ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -294,3 +295,41 @@ def ocs_install_verification(timeout=600, skip_osd_distribution_check=False):
                 item for item in crush_rule['steps'] if item.get('type') == 'zone'
             ], f"{crush_rule['rule_name']} is not with type as zone"
         log.info("Verified - pool crush rule is with type: zone")
+
+
+def add_capacity(capacity):
+    """
+   Add storage capacity to the cluster
+   Args:
+        capacity(int): Size of the storage to add as number of deviceSets
+   Returns:
+        boolean : Returns True if all OSDs are in Running state
+   """
+    ocp = OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE, kind=constants.STORAGECLUSTER)
+    sc = ocp.get()
+    device_set_count = sc.get('items')[0].get('spec').get('storageDeviceSets')[0].get('count')
+    capacity_to_add = device_set_count + capacity
```

Comment:
As mentioned in earlier comments in old PRs, the formula to derive the storageDeviceSets->count to add additional capacity is as follows:
storageDeviceSets->count = (capacity requested / osd capacity ) + existing count storageDeviceSets

I don't see that here. 

So, If i pass 512 GiB as int (i.e., just 512), then from your code, capacity_to_add  will have 512 + device_set_count. When this is 'patched', it would result in 512+device_set_count number of osds getting added to the system. Isn't it?

Can you please check and change the code accordingly?

Reply:
the code was changed that the function will get number of deviceSets to add instead of capacity in GB/TB

Reply:
We would like to keep add_capacity as generic as possible. It should work with both 4.2 and 4.3. From user point of view, the add_capacity needs to accept the capacity as a parameter. User does not have to know about osds and deviceSetcount and calculate them before calling add_capacity function. So, I would like this function to accept 'capacity to add' as parameter to it and use the formula internally to know the actual count of storagedevicesets to be 'patched' to meet the 'capacity to add'.
Please contact me or Shreekara if you need any clarification on this. Thanks!